client/src/components/SpotEntryForm/SpotEntryForm.js

```diff
@@ -0,0 +1,55 @@
+import styled from "@emotion/styled";
+import React from "react";
+import RedButton from "../RedButton/RedButton";
+import { useForm } from "react-hook-form";
+import { createSpot } from "../../api/spots";
+import PropTypes from "prop-types";
+
+const EntryForm = styled.form`
+  label,
+  input,
+  textarea {
+    font-family: var(--font-lato);
+    margin: 0.5rem 0;
+    display: block;
+    width: 100%;
+  }
+
+  button {
+    margin-top: 1.2rem;
+  }
+`;
+
+const SpotEntryForm = ({ location }) => {
+  const { register, handleSubmit } = useForm();
+
+  const onSubmit = (data) => {
+    console.log(data);
+    try {
+      data.latitude = location.latitude;
+      data.longitude = location.longitude;
+      const spotCreated = createSpot(data);
+      console.log(spotCreated);
+    } catch (error) {
+      console.error(error);
+    }
+  };
+  return (
+    <EntryForm onSubmit={handleSubmit(onSubmit)}>
+      <label htmlFor="title">Título</label>
+      <input name="title" required ref={register} />
+      <label htmlFor="description"> Comentarios </label>
+      <textarea name="description" rows={3} ref={register}></textarea>
+      <label htmlFor="visitDate">Día de visita</label>
+      <input name="visitDate" type="date" ref={register} />
+      <RedButton>CREAR SPOT</RedButton>
```

Comment:
A component name will give you information about the structure and features, not about the styling like the color.
It's similar to HTML elements.

Instead of calling a button `RedButton`,  you would call it like `SubmitButton`, `FormButton`, `Button` or `CloseButton` (examples). Take a look at UI frameworks like Material-UI or Bootstrap for more examples.
Imagine, that you create a second theme where you change the color of this button from red to blue. Now you would have a button called `RedButton` which is blue. 

This is a generial issue in your project. Your CSS variables (colors) shouldn't say what the color is, instead, it should say where to use it. 
Thus, `--color-darkgrayborder` is not a good name. Better: `--color-border` or `--color-primary-border` or `--color-modal-border`.
Otherwise it's not possible to have multiple themes.